core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -108,6 +108,29 @@ public static DataFile fromManifest(ManifestFile manifest) {
         .build();
   }
 
+  public static DataFile fromPositionDelete(DeleteFile deleteFile, PartitionSpec spec) {
```

Comment:
Do we need an assert here to make sure someone doesn't pass through an Equality delete?

Reply:
In the end, decided to make the method work for both kind of delete files, as the DeleteFile object should not be different:  https://github.com/apache/iceberg/pull/4812#discussion_r918441075